src/api/app/models/bs_request.rb

```diff
@@ -62,6 +64,17 @@ def save!
     notify if new
   end
 
+  def self.find_by_number!(number)
```

Comment:
hm, I could change the exception, but that would mean that I need to catch that on in all api cases and raise the other one to get a proper xml status error...